sources/api/bootstrap-containers/src/main.rs

```diff
@@ -518,38 +519,33 @@ where
 /// Handles the `mark-bootstrap` subcommand, which is called by the bootstrap
 /// container's systemd unit, which could potentially cause a concurrent invocation
 /// in this binary after the API setting finalizes.
-async fn mark_bootstrap<P>(args: MarkBootstrapArgs, socket_path: P) -> Result<()>
-where
-    P: AsRef<Path>,
-{
+fn mark_bootstrap(args: MarkBootstrapArgs) -> Result<()> {
     let container_id: &str = args.container_id.as_ref();
     let mode = args.mode.as_ref();
     info!("Mode for '{}' is '{}'", container_id, mode);
 
     // When 'mode' is 'once', the container is marked as 'off' once it
-    // finishes. This guarantees that the the container is only started in
+    // finishes. This guarantees that the container is only started in
     // the boot where it was created.
     if mode != "always" {
-        let formatted = format!("settings.bootstrap-containers.{}.mode", container_id);
-        let key = Key::new(KeyType::Data, &formatted)
-            .context(error::KeyFormatSnafu { key: formatted })?;
-        let value = serialize_scalar(&"off".to_string()).context(error::SerializeSnafu)?;
-
-        let mut map = HashMap::new();
-        map.insert(key, value);
-        let settings: model::Settings = datastore::deserialization::from_map(&map)
-            .context(error::SettingsDeserializeSnafu { settings: map })?;
-
+        let formatted = format!("settings.bootstrap-containers.{}.mode=off", container_id);
         info!("Turning off container '{}'", container_id);
-        apiclient::set::set(socket_path, &settings)
-            .await
+        let status = Command::new("apiclient")
```

Comment:
There is a helper method in this crate `command`, wouldn't it work for your case?